widget/embedded/src/components/Quote/QuoteTrigger .tsx

```diff
@@ -0,0 +1,161 @@
+import type { QuoteTriggerProps } from './Quote.types';
+import type { ReactNode } from 'react';
+
+import { i18n } from '@lingui/core';
+import {
+  ChevronDownIcon,
+  ChevronRightIcon,
+  Divider,
+  Image,
+  Tooltip,
+  Typography,
+} from '@rango-dev/ui';
+import React from 'react';
+
+import { getContainer } from '../../utils/common';
+
+import {
+  IconContainer,
+  ImageContainer,
+  MoreStep,
+  Row,
+  Trigger,
+} from './Quote.styles';
+
+const MAX_STEPS = 4;
+const INDEX_OF_STEP = 3;
+
+const ImageComponent = (props: {
+  content: ReactNode;
+  state?: 'error' | 'warning' | undefined;
+  src: string;
+  open?: boolean;
+  className?: string;
+}) => {
+  const tooltipContainer = getContainer();
+
+  return (
+    <Tooltip
+      container={tooltipContainer}
+      side="bottom"
+      sideOffset={4}
+      open={props.open}
+      content={props.content}>
+      <ImageContainer className={props.className} state={props.state}>
+        <Image src={props.src} size={16} />
+      </ImageContainer>
+    </Tooltip>
+  );
+};
+
+export function QuoteTrigger(props: QuoteTriggerProps) {
+  const { quoteRef, recommended, setExpanded, steps, numberOfSteps, expanded } =
+    props;
+  const tooltipContainer = getContainer();
+
+  return (
+    <Trigger
+      ref={(ref) => (quoteRef.current = ref)}
+      recommended={recommended}
+      onClick={() => setExpanded((prevState) => !prevState)}>
+      <Row>
+        <Typography variant="body" size="xsmall">
+          {i18n.t('Via:')}
+        </Typography>
+        <Divider direction="horizontal" size={4} />
+        {steps.map((step, index) => {
+          const key = `item-${index}`;
+          const arrow = (
+            <IconContainer>
+              <ChevronRightIcon size={12} color="black" />
+            </IconContainer>
+          );
+
+          return (
+            <React.Fragment key={key}>
+              {numberOfSteps <= MAX_STEPS ||
+              (numberOfSteps > MAX_STEPS && index < INDEX_OF_STEP) ? (
```

Comment:
NIT: I think it is better to use MAX_STEPS - 1 instead of INDEX_OF_STEP  Or with a more readable name

Reply:
fixed